chsdi/templates/htmlpopup/shop_product.mako

```diff
@@ -3,29 +3,41 @@
 <%def name="table_body(c, lang)">
 
 <%
+    import requests
+    layer = c['layerBodId']
     lang = lang if lang in ('fr','it','en') else 'de'
     fid = c['featureId']
     webDavHost = request.registry.settings['webdav_host']
     image = webDavHost + '/swisstopoproducts/250/' + fid + '.jpg'
+    r = requests.get(image)
+    r = r.status_code
+    shopUrl = h.order_mapsheet(lang)
     name = 's_title_%s' % lang
     if c['attributes']['scale']:
         c['attributes']['scale'] = h.format_scale(c['attributes']['scale'])
-    if c['attributes']['price']:
-        c['attributes']['price'] = h.format_price(c['attributes']['price'])
     attr = []
     attr_poss = ['number', 'name', 'scale', 'release', 'data', 'isbn', 'author', 'price']
     for ap in attr_poss:
         if ap in c['attributes']:
             if c['attributes'][ap]:
                 attr.append(ap)
+    if 'price' in attr:
+        c['attributes']['price'] = h.format_price(c['attributes']['price'])
     rowspan = len(attr) + 1
+    if r != 404:
```

Comment:
here I guess you want to check if the resource exists. Good news we have a helper function for that.
use
```python
h.resource_exists(image)
```
as we could have something else than a 404 error and still not have the image.